homeassistant/config_entries.py

```diff
@@ -2761,11 +2761,29 @@ def async_update_reload_and_abort(
         unique_id: str | None | UndefinedType = UNDEFINED,
         title: str | UndefinedType = UNDEFINED,
         data: Mapping[str, Any] | UndefinedType = UNDEFINED,
+        data_updates: dict[str, Any] | UndefinedType = UNDEFINED,
         options: Mapping[str, Any] | UndefinedType = UNDEFINED,
         reason: str | UndefinedType = UNDEFINED,
         reload_even_if_entry_is_unchanged: bool = True,
     ) -> ConfigFlowResult:
-        """Update config entry, reload config entry and finish config flow."""
+        """Update config entry, reload config entry and finish config flow.
+
+        :param data: used to replace the entry data with new data
+        :param data_updates: used to merge of updated data with pre-existing entry data
+        :param options: used to replace the entry options with new options
+        :param title: used to replace the title of the entry
+        :param unique_id: used to replace the unique_id of the entry
+
+        :param reason: used to set the reason for the abort, defaults to
+        `reauth_successful` or `reconfigure_successful` based on flow source
+
+        :param reload_even_if_entry_is_unchanged: set this to `False` if the entry
+        should not be reloaded if it is unchanged
+        """
+        if data_updates is not UNDEFINED:
+            if data is not UNDEFINED:
+                raise ValueError("Cannot set both data and data_updates")
+            data = {**entry.data, **data_updates}
```

Comment:
How about modern syntax?
```suggestion
            data = entry.data | data_updates
```